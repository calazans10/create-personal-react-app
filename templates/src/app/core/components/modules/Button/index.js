import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './index.scss';

const Button = ({ type, children, onClick }) => {
  const btnClasses = classNames({
    button: true,
    [`button--${type}`]: true,
  });

  return (
    <button type="button" className={btnClasses} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(['default', 'primary', 'success', 'danger', 'warning']),
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  type: 'default',
};

export default Button;
