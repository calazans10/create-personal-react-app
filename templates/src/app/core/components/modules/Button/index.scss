@import 'styles/utilities';

.button {
  display: block;
  width: 100%;
  margin-bottom: 0;
  padding: rem(8) rem(4);
  border: none;
  border-radius: rem(5);
  border-color: transparent;
  background-image: none;
  color: #fff;
  font-size: em(14);
  font-weight: normal;
  line-height: 1.3333333;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  touch-action: manipulation;
  user-select: none;

  &--default {
    background-color: #bdc3c7;
  }

  &--primary {
    background-color: #00aeef;
  }

  &--success {
    background-color: #21d376;
  }

  &--danger {
    background-color: #e33b15;
  }

  &--warning {
    background-color: #f39c12;
  }
}
