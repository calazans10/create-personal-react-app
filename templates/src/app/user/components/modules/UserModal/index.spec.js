import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../../../../core/components/modules/Button';
import { UserModal } from './index';

describe('UserModal', () => {
  const props = {
    user: {
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz',
      address: {
        street: 'Kulas Light',
        suite: 'Apt. 556',
        city: 'Gwenborough',
        zipcode: '92998-3874',
        geo: {
          lat: '-37.3159',
          lng: '81.1496',
        },
      },
      phone: '1-770-736-8031 x56442',
      website: 'hildegard.org',
      company: {
        name: 'Romaguera-Crona',
        catchPhrase: 'Multi-layered client-server neural-net',
        bs: 'harness real-time e-markets',
      },
    },

    onClearUser: jest.fn(),
    onRequestDeleteUser: jest.fn(),
    onHideModal: jest.fn(),
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<UserModal {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle the close modal event', () => {
    const wrapper = shallow(<UserModal {...props} />);
    wrapper
      .find(Button)
      .at(0)
      .simulate('click');
    expect(props.onClearUser).toHaveBeenCalledTimes(1);
    expect(props.onHideModal).toHaveBeenCalledTimes(1);
  });

  it('should handle the click delete user event', () => {
    const wrapper = shallow(<UserModal {...props} />);
    wrapper
      .find(Button)
      .at(1)
      .simulate('click');
    expect(props.onRequestDeleteUser).toHaveBeenCalledTimes(1);
  });
});
