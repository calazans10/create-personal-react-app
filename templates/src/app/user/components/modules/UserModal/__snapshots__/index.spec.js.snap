// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModal renders without crashing 1`] = `
<Connect(Modal)
  onHideModal={[Function]}
>
  <div
    className="user-modal"
  >
    <h2
      className="user-modal__title"
    >
      Remover usuário
    </h2>
    <p
      className="user-modal__caption"
    >
      Tem certeza que deseja remover o usuário 
      <strong>
        Bret
      </strong>
      ?
    </p>
    <div
      className="user-modal__actions"
    >
      <Button
        onClick={[Function]}
        type="default"
      >
        Cancelar
      </Button>
      <Button
        onClick={[Function]}
        type="danger"
      >
        Sim
      </Button>
    </div>
  </div>
</Connect(Modal)>
`;
